account_asset_date/models/account_asset.py

```diff
@@ -0,0 +1,152 @@
+            self.write({'depreciation_line_ids': commands})
+            return True
+
+        amount_to_depr = residual_amount = self.value_residual
+        if (posted_deprec_line_ids and
+                posted_deprec_line_ids[-1].depreciation_date):
+            depreciation_date = (datetime.strptime(
+                posted_deprec_line_ids[-1].depreciation_date,
+                DF).date() + relativedelta(
+                months=+self.method_period))
+        else:
+            depreciation_date = datetime.strptime(
+                self._get_last_depreciation_date()[self.id], DF).date() if (
+                    self.prorata) else (datetime.strptime(
+                        self._start_depreciation_date(
+                            self.start_depreciation[:4] + '-01-01' if (
+                                self.method_period >= 12) else
+                            self.start_depreciation), DF).date())
+
+        day = depreciation_date.day
+        month = depreciation_date.month
+        year = depreciation_date.year
+        total_days = (year % 4) and 365 or 366
+
+        undone_dotation_number = self._compute_board_undone_dotation_nb(
+            depreciation_date, total_days)
+
+        for seq in range(len(posted_deprec_line_ids),
+                         undone_dotation_number):
+            sequence = seq + 1
+            amount = self._compute_board_amount(
+                sequence, residual_amount, amount_to_depr,
+                undone_dotation_number, posted_deprec_line_ids,
+                total_days, depreciation_date)
+            amount = self.currency_id.round(amount)
+            if float_is_zero(
+                    amount, precision_rounding=self.currency_id.rounding):
+                continue
+            residual_amount -= amount
+            vals = {
+                'amount': amount,
+                'asset_id': self.id,
+                'sequence': sequence,
+                'name': (self.code or '') + '/' + str(sequence),
+                'remaining_value': residual_amount,
+                'depreciated_value': self.value - (
+                    self.salvage_value + residual_amount),
+                'depreciation_date': depreciation_date.strftime(DF),
+            }
+            commands.append((0, False, vals))
+            # Considering Depr. Period as months
+            depreciation_date = date(year, month, day) + relativedelta(
+                months=+self.method_period)
+            day = depreciation_date.day
+            month = depreciation_date.month
+            year = depreciation_date.year
+
+        self.write({'depreciation_line_ids': commands})
+
+        return True
+
+    @api.multi
+    def _get_last_depreciation_date(self):
+        if not self.different_dates:
+            return super(AccountAssetAsset, self)._get_last_depreciation_date()
+
+        self.env.cr.execute("""
+            SELECT a.id as id,
+              COALESCE(MAX(m.date),a.start_depreciation) AS date
```

Comment:
Sorry, @lescobarvx  I didn't see this before.
The date returned by this query is in UTC. Is there no problem?

cc. @dsabrinarg

Reply:
@josemoralesp 

I copied from original method to overwrite it and they use that query

https://github.com/odoo/odoo/blob/11.0/addons/account_asset/models/account_asset.py#L123-L129